app/src/main/java/com/nginx/image/core/PhotoTransformer.java

```diff
@@ -0,0 +1,144 @@
+                // THis is returned here because the image doesn't need to be reoriented at all.
+            }
+            AffineTransform t = getExifTransformation(new ImageInformation(orientation,width,height));
+            bufferedImage = transformImage(bufferedImage,t);
+            return bufferedImage;
+        } catch (Exception e) {
+            LOGGER.error("This is the general exception message: ", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Transforms an image using {@link AffineTransformOp} and {@link Graphics2D}
+     *
+     * @param image the image to transform {@link BufferedImage}
+     * @param transform the {@link AffineTransform} object
+     *
+     * @return a {@link BufferedImage} object
+     */
+    private static BufferedImage transformImage(BufferedImage image, AffineTransform transform) {
+        AffineTransformOp op = new AffineTransformOp(transform, AffineTransformOp.TYPE_NEAREST_NEIGHBOR);
+
+        BufferedImage destinationImage = op.createCompatibleDestImage(image, (image.getType() == BufferedImage.TYPE_BYTE_GRAY) ? image.getColorModel() : null);
+        Graphics2D g = destinationImage.createGraphics();
+        g.setBackground(Color.WHITE);
+        g.clearRect(0, 0, destinationImage.getWidth(), destinationImage.getHeight());
+        destinationImage = op.filter(image, destinationImage);
+        return destinationImage;
+    }
+
+    /**
+     * Static method that creates and returns a new {@link AffineTransform} object based on
+     * the info param
+     *
+     * @param info an {@link ImageInformation} object
+     *
+     * @return an {@link AffineTransform} object
+     */
+    private static AffineTransform getExifTransformation(ImageInformation info) {
+        AffineTransform t = new AffineTransform();
+
+        switch (info.orientation) {
+            case 1:
+                break;
+            case 2: // Flip X
+                t.scale(-1.0, 1.0);
+                t.translate(-info.width, 0);
+                break;
+            case 3: // PI rotation
+                t.translate(info.width, info.height);
+                t.rotate(Math.PI);
+                break;
+            case 4: // Flip Y
+                t.scale(1.0, -1.0);
+                t.translate(0, -info.height);
+                break;
+            case 5: // - PI/2 and Flip X
+                t.rotate(-Math.PI / 2);
+                t.scale(-1.0, 1.0);
+                break;
+            case 6: // -PI/2 and -width
+                t.translate(info.height, 0);
+                t.rotate(Math.PI / 2);
+                break;
+            case 7: // PI/2 and Flip
+                t.scale(-1.0, 1.0);
+                t.translate(-info.height, 0);
+                t.translate(0, info.width);
+                t.rotate(  3 * Math.PI / 2);
+                break;
+            case 8: // PI / 2
+                t.translate(0, info.width);
+                t.rotate(  3 * Math.PI / 2);
+                break;
```

Comment:
It's good practice to have a default case in a switch statement. Can we be 100% certain that the value will always be one of those covered by the case statements?

Reply:
I agree with @cpretzer . We should have a default case.

Reply:
good call, and easily done